src/tsung/ts_mqtt.erl

```diff
@@ -80,9 +80,9 @@ dump(A, B) ->
 get_message(#mqtt_request{type = connect, clean_start = CleanStart,
                           keepalive = KeepAlive, will_topic = WillTopic,
                           will_qos = WillQos, will_msg = WillMsg,
-                          will_retain = WillRetain, username = UserName, password = Password},
+                          will_retain = WillRetain, username = UserName,
+                          password = Password, client_id = ClientId},
             #state_rcv{session = MqttSession}) ->
-    ClientId = ["tsung-", ts_utils:randombinstr(10)],
```

Comment:
I'm not very deeply familiar with MQTT, but if client ID is required, we could generate it in case it is not provided.

Reply:
I had thought about that, but it seems that anyone using the protocol may quickly generate it using Tsung facilities. 

Do you reckon it would be better to have another function clause which automatically generates the Client ID, when it's not provided?

Reply:
Like I said, if this is required, then I'd say yes. Otherwise this would be a breaking change.

Reply:
I've submitted a new commit which hopefully takes care of this. However, I'll need someone to validate it, since my Erlang foo is quite low.

Reply:
We also added two unit tests scenarios, providing a custom client ID or using the default implementation. So far it does not seem to be a breaking change.